### PR TITLE
Video Compression toggling/options

### LINKED ISSUE #112: Add in levels of compression
Add in a variety of options to allow users to compress their media at different levels.

Ideal hours: 10
Related story: #97

### PR DESCRIPTION
**Pull request by BDSchoeler**
Pull request for #97 
Added option to change level of compression and to disable or enable it depending on wifi connection

Includes unit tests and ui test.
close #112 #113 #114

### CODE REVIEW

File: src/org/thoughtcrime/securesms/mms/MediaConstraints.java [hunk 1]
```diff
@@ -28,6 +29,16 @@
...
 
+  private static final int LOW_VIDEO_COMPRESSION_HEIGHT  = 1080;
+  private static final int LOW_VIDEO_COMPRESSION_WIDTH  = 1920;
+  private static final int LOW_VIDEO_COMPRESSION_BITRATE = 750000;
+  private static final int MEDIUM_VIDEO_COMPRESSION_HEIGHT  = 720;
+  private static final int MEDIUM_VIDEO_COMPRESSION_WIDTH  = 1280;
+  private static final int MEDIUM_VIDEO_COMPRESSION_BITRATE = 600000;
+  private static final int HIGH_VIDEO_COMPRESSION_HEIGHT  = 480;
+  private static final int HIGH_VIDEO_COMPRESSION_WIDTH  = 640;
+  private static final int HIGH_VIDEO_COMPRESSION_BITRATE =  450000;
```
Comment (benrs) [inline src/org/thoughtcrime/securesms/mms/MediaConstraints.java:40]: Maybe shove these in a map-like structure instead so adding new ones is easy?

File: test/androidTest/java/org/thoughtcrime/securesms/VideoCompressionOptionMenuTest.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    public ActivityTestRule<ConversationListActivity> mActivityTestRule = new ActivityTestRule<>(ConversationListActivity.class);
+
+    @Test
+    public void videoCompressionOptionMenuTest() {
+        //open action bar menu
+        openActionBarOverflowOrOptionsMenu(getInstrumentation().getTargetContext());
+        //open settings tab
+        openSettingsMenu();
+        //open chat and media settings tab
+        openChatAndMediaMenu();
```
Comment (benrs) [inline test/androidTest/java/org/thoughtcrime/securesms/VideoCompressionOptionMenuTest.java:46]: Should think of using the util classes/moving these to the util classes.

File: src/org/thoughtcrime/securesms/mms/MediaConstraints.java (hunk 1 above)
Comment (BDSchoeler) [reply src/org/thoughtcrime/securesms/mms/MediaConstraints.java:40]: I'm not a biggest fan of having it his way, but in all the classes that are children of this one, they have lists of constants like this, so i wanted to proceed with similar practices.

File: test/androidTest/java/org/thoughtcrime/securesms/VideoCompressionOptionMenuTest.java (hunk 2 above)
Comment (BDSchoeler) [reply test/androidTest/java/org/thoughtcrime/securesms/VideoCompressionOptionMenuTest.java:46]: good idea

Comment (BDSchoeler) [reply test/androidTest/java/org/thoughtcrime/securesms/VideoCompressionOptionMenuTest.java:46]: @benrs Is there an util class that already has general helper methods for expresso?

Comment (benrs) [reply test/androidTest/java/org/thoughtcrime/securesms/VideoCompressionOptionMenuTest.java:46]: All the util classes are under the util folder, if you can't find what you need there then no

Comment (claudds) [reply test/androidTest/java/org/thoughtcrime/securesms/VideoCompressionOptionMenuTest.java:46]: @BDSchoeler  check the ConversationActions class

File: test/unitTest/java/org/thoughtcrime/securesms/mms/VideoCompressionOptionsTest.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+import static org.mockito.Mockito.times;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(TextSecurePreferences.class)
+public class VideoCompressionOptionsTest extends BaseUnitTest {
```
Comment (claudds) [inline test/unitTest/java/org/thoughtcrime/securesms/mms/VideoCompressionOptionsTest.java:24]: Is it necessary to have this extend the BaseUnitTest class? don't know if the other test we've written have had the need to do this

**Review by claudds — APPROVED** [review]
Good to merge after checking some of the tests

File: test/unitTest/java/org/thoughtcrime/securesms/mms/VideoCompressionOptionsTest.java (hunk 3 above)
Comment (BDSchoeler) [reply test/unitTest/java/org/thoughtcrime/securesms/mms/VideoCompressionOptionsTest.java:24]: It's not necessary, but base unit test does some mocking/static mocking for you, which reduces repeated code.

**Review by samantha-kerr — APPROVED** [review]
Code and tests look good

File: test/androidTest/java/org/thoughtcrime/securesms/VideoCompressionOptionMenuTest.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
+package org.thoughtcrime.securesms;
+
+
```
Comment (lori-dalkin) [inline test/androidTest/java/org/thoughtcrime/securesms/VideoCompressionOptionMenuTest.java:3]: Remove the extra line

**Review by lori-dalkin — APPROVED** [review]
Looks good to me! Good test coverage.

**Review by jacobrs — APPROVED** [review]
Good once codacy fixes applied!

**Pull request merged**